Merging headers into one

marcosmarp opened the pull request:
Solves #12

marcosmarp commented on main.h:
```diff
@@ -0,0 +1,571 @@
+
+};
+
+std::vector<userData> usersLoginInfo;
```
I know that variables shouldn't be declared or defined in headers, only in cpp files, but i can't really use any function defined here without declaring the vector here; maybe another possible issue to be solved could be passing the vector as an argument to every function instead of the current method?

nahueespinosa replied:
Yeah, that's why you should really move all the function and variable definitions into `main.cpp`. 😉

nahueespinosa commented on main.h:
```diff
@@ -0,0 +1,571 @@
+/* Functions definitions */
+
+/* Stetic functions*/
+void outputThreeDots(){ 
```
This is a function definition, it should go in `main.cpp`.

nahueespinosa commented on main.h:
```diff
@@ -0,0 +1,571 @@
+
+}
+
+void outputTab(int numberOfTabs) {
```
Same for this, and the following function definitions.

nahueespinosa commented on main.cpp:
```diff
@@ -1,5 +1,7 @@
-#include "functions_declarations.h"
+#include "main.h"
+
+extern std::vector<userData> usersLoginInfo;
+extern unsigned long numberOfUsers;
```
You won't need these when all the definitions land in this file,

nahueespinosa commented on main.h:
```diff
@@ -0,0 +1,571 @@
+/* Functions declarations */
+
+/* Stetics functions */
+void outputThreeDots(); // Writes in console three dots (.) with .5 seconds of delay between them
+void outputTab(int); // Writes in console the user specified amount of tab spaces
+
+/* Base functions */
+void registerNewUser(); // The whole of the registration process
+void logIn(); // The login process
+
+/* Last level functions */
+void recoverPassword_questionMethod(unsigned long); // Goes trough the validation process of the user's security answer so he can recover the password.
+bool checkForEmailDuplicate(std::string); // Checks if the submitted email in the registration process is already in use
+bool checkForUsernameDuplicate(std::string); // Checks if the submitted username in the registration process is already in use
+void newPassword(); // Process to upload a new password to an user
+void enterPassword(unsigned long); // Process to enter the password in the login system
+unsigned long searchUsername(std::string); // Searchs for the presence of the submitted username in the login process
+bool validatePassword(std::string, unsigned long); // Validates the submitted password in the login process
+bool validateSecAnswer(std::string, unsigned long); // Validates a security answer
+void generateNewPassword(unsigned long); // Replaces the already saved password of an user with a new one
```
These function declarations are correctly placed.

nahueespinosa reviewed (changes requested):
Left some comments!

marcosmarp commented:
Just implemented your corrections! No more external variables and the definitions are now in the .cpp file

nahueespinosa commented on main.h:
```diff
@@ -1,20 +1,39 @@
-bool validateSecAnswer(std::string, unsigned long); // Validates a security answer
-void generateNewPassword(unsigned long); // Replaces the already saved password of an user with a new one
+#pragma once
+#include <string>
+#include <cstdlib>
+#include <iostream>
+#include <stdlib.h>
+#include <vector>
+#include <conio.h>
+#include <stdio.h>
```
Please, remove the headers you don't need. For starters, all the ones ending in `.h` (C standard library headers).

And move `#include <vector>` and `#include <iostream>` to `main.cpp` as you don't need them for any declaration here.

nahueespinosa reviewed (changes requested):
Cool! One last thing. You don't want to include files you are not using, specially in a header file that's going to be included in other programs potentially.

A good rule of thumb is that you should keep all the includes in your `.cpp` file unless you need them for function or variable declarations in the header.

marcosmarp commented:
Done! Deleted unnecessary libraries and moved every one remaining to the .cpp (except string, that was necessary in de declarations)

>Please, remove the headers you don't need. For starters, all the ones ending in .h (C standard library headers).

I had to leave <conio.h> included, i use _getch() to receive the password from the user, mainly because is the only function i could find that doesn't show an echo in the console of the user's input (so nobody can see in the screen what his password is)

nahueespinosa reviewed (approved):
> I had to leave <conio.h> included

Got it, we'll review that later. Approved!
